### ISSUE TITLE
The test results were all black.

### ISSUE THREAD

**Issue by DrewdropLife** [ehsanik/SeGAN#5]
Hello! When I used the pretrained model provided by you, the test results are all normal in visualization, but if I start training from scratch, or continue training on the basis of your pretrained model, the test results are all black, and the loss of texture task is all nan.
The commands I use are as follows.

> th main.lua -weights_segmentation "weights/segment_gt_sv" -end2end -weights_texture "weights/texture_gt_sv" -istrain "test"

When I trained from scratch and then test, I only changed the paths in the above code.

I added the visual code save_img() in the function test_texture() to visualize the pred_center.( Because I use the function test_end2end() in main.lua instead of the function test() in GANTextureNet.lua )

This is the first question. 
Besides, I want to ask you some other questions for help.  
          
2. What is the 'cvpr' parameter mean?            
3. I want to train on the natural image, but I can't get the RGB image of the full object. What's the solution in this case?
4. I found that the accuracy of the segmentation of visible regions in the paper is only 68, Is the precision drop too big? What is the reason? In Table 1. Segmentation evaluation.

Thank you very much for your help!

**Comment by DrewdropLife** [comment]
For the first question, I can confirm that there is a problem with the saved texture-network. Because visualization during the training phase seems normal, but the output of netG in the testing phase changes to 'nan'. 
I haven't changed the code that save the network, and don't know how to solve it.

**Comment by ehsanik** [comment]
1. If the visualization are okay after the first reloading of the pretrained model and then becomes nan I would say there is an issue with how you train your model. It can be either with the data, or maybe learning rate (you should lower the learning rate if you are not training from scratch or some other problem). 
2. The cvpr flag is always True. We had different models before releasing the code and we used this flag to enable the model we used for our CVPR submission. It is basically redundant now.
3.  Unfortunately there is no way to train this model if you do not have the full object. My only suggestion is, if you can at least get the full segmentation (not the texture), freeze the texture generation network and just fine tune the segmentation part. Other wise you can just test this model on your natural images dataset. I don't think there is any other solutions for training if you don't have the labels. 
4. This is the intersection over union of the predicted visible segmentation mask with ground truth visible mask, which is basically the instance segmentation task and it is not a trivial problem. Refer to 

> [51] S. Zagoruyko, A. Lerer, T.-Y. Lin, P. O. Pinheiro, S. Gross,
S. Chintala, and P. Dollar. A multipath network for object ´
detection. In BMVC, 2016

**Comment by DrewdropLife** [comment]
Thank you for your detailed reply. I'll take a closer look at the problem of model training.
